Terminal.Gui/ConsoleDrivers/CursesDriver/CursesDriver.cs

```diff
@@ -492,8 +502,19 @@ void GetEscSeq (ref int code, ref Key k, ref int wch2, ref KeyEvent key, ref Con
 			}
 		}
 
+		MouseFlags lastMouseFlags;
```

Comment:
in the v2 codebase, I refactored this as follows to make it more readable. Do you agree it's more readable?

```cs
	MouseFlags _lastMouseFlags;

	void ProcessMouseEvent (MouseFlags mouseFlag, Point pos)
	{
		bool WasButtonReleased (MouseFlags flag)
		{
			return flag.HasFlag (MouseFlags.Button1Released) ||
				flag.HasFlag (MouseFlags.Button2Released) ||
				flag.HasFlag (MouseFlags.Button3Released) ||
				flag.HasFlag (MouseFlags.Button4Released);
		}

		bool IsButtonNotPressed (MouseFlags flag)
		{
			return !flag.HasFlag (MouseFlags.Button1Pressed) &&
				!flag.HasFlag (MouseFlags.Button2Pressed) &&
				!flag.HasFlag (MouseFlags.Button3Pressed) &&
				!flag.HasFlag (MouseFlags.Button4Pressed);
		}

		bool IsButtonClickedOrDoubleClicked (MouseFlags flag)
		{
			return flag.HasFlag (MouseFlags.Button1Clicked) ||
				flag.HasFlag (MouseFlags.Button2Clicked) ||
				flag.HasFlag (MouseFlags.Button3Clicked) ||
				flag.HasFlag (MouseFlags.Button4Clicked) ||
				flag.HasFlag (MouseFlags.Button1DoubleClicked) ||
				flag.HasFlag (MouseFlags.Button2DoubleClicked) ||
				flag.HasFlag (MouseFlags.Button3DoubleClicked) ||
				flag.HasFlag (MouseFlags.Button4DoubleClicked);
		}

		if ((WasButtonReleased (mouseFlag) && IsButtonNotPressed (_lastMouseFlags)) ||
			(IsButtonClickedOrDoubleClicked (mouseFlag) && _lastMouseFlags == 0)) {
			return;
		}

		_lastMouseFlags = mouseFlag;

		var me = new MouseEvent () {
			Flags = mouseFlag,
			X = pos.X,
			Y = pos.Y
		};
		_mouseHandler (me);
	}
```

Reply:
Yes I agree it's much more readable. Thanks. The only thing I didn't the same as yours was not using the underscore because it's a rule implemented for the v2.

Done with https://github.com/gui-cs/Terminal.Gui/pull/2786/commits/320dcda3ce2929b29a0b53da5e32f50d574c6db2.